octobot_trading/modes/abstract_trading_mode.py

```diff
@@ -117,23 +123,73 @@ def is_backtestable() -> bool:
+        """
+        Stops all producers and consumers
+        """
         for producer in self.producers:
             await producer.stop()
         for consumer in self.consumers:
             await consumer.stop()
         self.exchange_manager = None
 
     async def create_producers(self) -> list:
-        raise NotImplementedError("create_producers not implemented")
+        """
+        Creates the instance of producers listed in MODE_PRODUCER_CLASSES
+        :return: the list of producers created
+        """
+        return [
+            await self._create_mode_producer(mode_producer_class)
+            for mode_producer_class in self.MODE_PRODUCER_CLASSES
+        ]
+
+    async def _create_mode_producer(self, mode_producer_class):
+        """
+        Creates a new :mode_producer_class: instance and starts it
+        :param mode_producer_class: the trading mode producer class to create
+        :return: the producer class created
+        """
+        mode_producer = mode_producer_class(
+            exchanges_channel.get_chan(constants.MODE_CHANNEL, self.exchange_manager.id),
+            self.config, self, self.exchange_manager)
+        await mode_producer.run()
+        return mode_producer
 
     async def create_consumers(self) -> list:
-        raise NotImplementedError("create_consumers not implemented")
+        """
+        Creates the instance of consumers listed in MODE_CONSUMER_CLASSES
+        :return: the list of consumers created
+        """
+        return [
+            await self._create_mode_consumer(mode_consumer_class)
+            for mode_consumer_class in self.MODE_CONSUMER_CLASSES
+        ]
+
+    async def _create_mode_consumer(self, mode_consumer_class):
+        """
+        Creates a new :mode_consumer_class: instance and subscribe this new consumer to the trading mode channel
+        :param mode_consumer_class: the trading mode consumer class to create
+        :return: the consumer class created
+        """
+        mode_consumer = mode_consumer_class(self)
+        await exchanges_channel.get_chan(constants.MODE_CHANNEL, self.exchange_manager.id).new_consumer(
+            consumer_instance=mode_consumer,
+            trading_mode_name=self.get_name(),
+            cryptocurrency=self.cryptocurrency if self.cryptocurrency else channel_constants.CHANNEL_WILDCARD,
+            symbol=self.symbol if self.symbol else channel_constants.CHANNEL_WILDCARD,
+            time_frame=self.time_frame if self.time_frame else channel_constants.CHANNEL_WILDCARD)
+        return mode_consumer
 
     def load_config(self) -> None:
+        """
+        Try to load TradingMode tentacle config. Calls set_default_config() if the tentacle config is empty
+        """
         # try with this class name
         self.trading_config = tentacles_manager_api.get_tentacle_config(self.exchange_manager.tentacles_setup_config,
                                                                         self.__class__)
```

Comment:
could we split this in 2 commits plz ? I'm not sure we will be able to find it later under a version commit

Reply:
:+1: